### ISSUE TITLE
local echos of redactions incorrectly hide state events

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#10691]
given redacting state events strips out the human visible content but not the structure of the event itself, they shouldn't be hidden from the timeline. e.g. redacting a user join event shouldn't hide their join from the timeline, just remove the displayname and avatar. in practice, disabling local echo of state event redactions would be a sufficient fix here

**Comment by ara4n** [comment]
this is really quite annoying, as it makes the timeline quite hard to follow when join/parts of an abusive user are redacted...

**Comment by lampholder** [comment]
Is this a local echo problem or a state-events-disappear-when-redacted problem?

As far as I can tell, redacting a state event has always (== pre-redesign) made it disappear from the UX entirely.

Would this be solved by showing a redacted representation of state events rather than erasing them completely?

**Comment by ara4n** [comment]
> Is this a local echo problem or a state-events-disappear-when-redacted problem?

the latter

> As far as I can tell, redacting a state event has always (== pre-redesign) made it disappear from the UX entirely.

I don't /think/ so, but i may be smoking crack. They used to remove the user-facing bits (i.e. displayname & avatar), but keep the semantic bits (i.e. `@foo:bar.com` joined the room))

> Would this be solved by showing a redacted representation of state events rather than erasing them completely?

Yes.

**Comment by turt2live** [comment]
> I don't /think/ so, but i may be smoking crack. They used to remove the user-facing bits (i.e. displayname & avatar), but keep the semantic bits (i.e. `@foo:bar.com` joined the room))

This has never been the case as far as I can tell. https://github.com/vector-im/riot-web/issues/5684 is evidence of this. 

Closing as a duplicate of https://github.com/vector-im/riot-web/issues/5684 and/or https://github.com/vector-im/riot-web/issues/3193
